### PR TITLE
Change password

### PR DESCRIPTION
**Pull request by smalishevskiy**
Fix issue #261

Fix problem with create XEP-0077. Implement more user frendly UI for change password

change_password

### CODE REVIEW

File: main/src/ui/manage_accounts/change_password_dialog.vala [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    
+        [GtkChild] private unowned Button change_password_button;
+        [GtkChild] private unowned Stack change_password_stack;
+        [GtkChild] private unowned Button cancel_button;
+        [GtkChild] private unowned Entry current_passwd_entry;
+        [GtkChild] private unowned Entry new_passwd_entry;
+        [GtkChild] private unowned Entry confirm_new_passwd_entry;
+        [GtkChild] private unowned Label change_password_error_label;
+
+        private bool are_forms_empty;
```
Comment (fiaxh) [inline main/src/ui/manage_accounts/change_password_dialog.vala:21]: This variable is set but never read.

**Review by fiaxh — COMMENTED** [review]
Thanks for your PR!

Instead of using a Grid-Layout with Labels and Entries, I'd suggest to use Adw.PasswordEntryRow (within an Adw.PreferenceGroup, whithin an Adw.PreferencePage). It gives us some password field features for free and would look like this (this is from the gnome settings password change dialog):

image

Please remove updates to the .po and .pot files, they make PRs harder to review. We'll update the files independently afterwards.

We try to get rid of references to the StreamInteractor from our UI code, and I'd prefer to not introduce any more in the meanwhile. Could you please create a PasswordChangeDialogModel, that has a function to change the password (basically doing the StreamInteraction calls there instead of in the UI code)? The dialog UI code should contain a reference to the model and call the respective function there.

As a general style remark, you abbreviate variable names in various places (passwd, pw, chpw, btn, s, a, ...). We typically don't do that in our codebase and I'd like to keep the variable naming style at least somewhat consistent.

**Comment by smalishevskiy** [comment]
> We try to get rid of references to the StreamInteractor from our UI code, and I'd prefer to not introduce any more in the meanwhile. Could you please create a PasswordChangeDialogModel, that has a function to change the password (basically doing the StreamInteraction calls there instead of in the UI code)? The dialog UI code should contain a reference to the model and call the respective function there.

Can you point me example of model in current code? That is not clean for me where I should create the model's class.

**Comment by fiaxh** [comment]
@smalishevskiy `Dino.Ui.ViewModel.PreferencesWindow` is an example, however the view model for the change password dialog would be much simpler. It could roughly look like this:
```
public class Dino.Ui.ViewModel.ChangePasswordDialog : Object {
    private StreamInteractor stream_interactor;
    public ChangePasswordDialog(StreamInteractor stream_interactor);
    public change_password(Account account, string password);
}
```
The UI would be given an instance of this class. When a new password was entered, the UI would call `change_password` on this class. `change_password` would make the call to the backend to change the password.

**Comment by fiaxh** [comment]
I implemented the changed that I've suggested and merged this in f8c004630f56914438fa1b114530f639748e41c1.

**Pull request closed without merging**